superset-frontend/src/components/Select/Select.tsx

```diff
@@ -416,6 +427,7 @@ const Select = ({
       allowNewOptions,
       initialOptions,
       isSingleMode,
+      onChange,
```

Comment:
@geido onChange was not on the list of dependencies in your original PR. It's a good idea to check that this function is being memoized 😉